pkg/cvo/cvo.go

```diff
@@ -269,6 +271,40 @@ func (optr *Operator) InitializeFromPayload(restConfig *rest.Config, burstRestCo
 	return nil
 }
 
+// ownerReferenceModifier sets the owner reference to the current CV resource if no other reference exists
+func (optr *Operator) ownerReferenceModifier(object metav1.Object) {
+	var found bool
+	// verify if there an existing owner reference which is not on a ClusterVersion resource
+	for _, ownerRef := range object.GetOwnerReferences() {
+		if ownerRef.Kind != "ClusterVersion" {
+			found = true
+			break
+		}
+	}
+	if found {
+		// If there are ownerReferences which for non ClusterVersion resource then remove all ClusterVersion
+		// ownerReferences to them
+		var nonCVORefs []metav1.OwnerReference
+		for _, ref := range object.GetOwnerReferences() {
+			if ref.Kind != "ClusterVersion" && ref.Name != optr.name {
```

Comment:
this seems a bit redundent with the earlier loop setting `found`.  How about:

```go
modified := false
var otherRefs []metav1.OwnerReference
for _, ref := range object.GetOwnerReferences() {
  if ref.Kind != "ClusterVersion" || ref.Name != optr.name) {
    otherRefs = append(otherRefs, ref)
  }
  if len(otherRefs) > 0 {
    // FIXME: Log that we are defering to these other owners, despite us continuing to semi-manage this resource
    object.SetOwnerReferences(otherRefs)
  } else {
    object.SetOwnerReferences([]metav1.OwnerReference{{
      APIVersion: configv1.GroupVersion.Identifier(),
      Kind:       "ClusterVersion",
      Name:       optr.name,
      UID:        optr.uid,
    }})
  }
}
```

Reply:
Updated as per suggestion.